components/authz-service/server/v2/authz.go

```diff
@@ -51,18 +56,35 @@ func (s *authzServer) IsAuthorized(
 func (s *authzServer) ProjectsAuthorized(
 	ctx context.Context,
 	req *api.ProjectsAuthorizedReq) (*api.ProjectsAuthorizedResp, error) {
+	// we check the version set in the channel on policy server
+	// in order to determine whether or not to filter projects for the request
+	version := s.vSwitch.Version
+	var projects []string
+	if version.Minor == api.Version_V0 {
+		// if IAM version is set to v2.0
+		// we override the requested projects because no filter should be applied on v2
+		projects = []string{}
+	} else {
+		projects = req.ProjectsFilter
+	}
 
 	projectsAuthorized, err := s.engine.V2ProjectsAuthorized(ctx,
 		engine.Subjects(req.Subjects),
 		engine.Action(req.Action),
 		engine.Resource(req.Resource),
-		engine.ProjectList(req.ProjectsFilter...))
+		engine.ProjectList(projects...))
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
-	// Generally we return the engine's response verbatim
-	// but there are two cases that need to be intercepted and adjusted.
-	if stringutils.SliceContains(projectsAuthorized, constants.AllProjectsID) {
+
+	if version.Minor == api.Version_V0 && len(projectsAuthorized) > 0 {
+		// if IAM version is set to v2.0
+		// as long as at least one project is authorized
+		// we override the filtered projects because no filter should be applied on v2
+		projectsAuthorized = []string{constants.AllProjectsExternalID}
+	} else if stringutils.SliceContains(projectsAuthorized, constants.AllProjectsID) {
+		// Generally we return the engine's response verbatim
+		// but there are two cases that need to be intercepted and adjusted.
 		if len(req.ProjectsFilter) == 0 {
 			// Engine allows all and we requested all, so signify it as all.
 			// This must be different than the requested notion of all,
```

Comment:
Not shown, but I found a minor bug that I introduced:
L97 contains a call to `s.logProjectQuery`. That would have likely been quite helpful to you during development--if I had put it in the right place! It should not be nested in an if-then-else, so please move it down two lines, to immediately before the `return` in L100. With that in the right place, running your battery of "use cases" in the PR preamble shows this in the supervisor logs (truncated to fit here). Note the `projects` shows the _requested projects_ and `result` shows the _allowed projects_.
```
projects="[automate auth not-real]" resource="iam:projects" result="[automate auth]" subject="[token:test]"
projects="[]" resource="system:status" result="[*]" subject="[team:local:admins user:local:admin]"
projects="[auth]" resource="iam:projects:auth" result="[auth]" subject="[token:test]"
projects="[automate]" resource="iam:projects:automate" result="[]" subject="[token:test]"
```